### DISCUSSION TITLE
How to integrate cucumberjs-allure2-reporter   into Protractor-Cucumber-Typescript Framework

### DISCUSSION THREAD

**Discussion by jobby-tech** [allure-framework/allure2#1333] [General Discussion]
Hi Team,

I have been using cucumber-html-reporter   for generating the reports, 

and looking to integrate cucumberjs-allure2-reporter   into Protractor-Cucumber-Typescript Framework

could you please help out on how to integrate?

Below is my existing Config File

export let config : Config ={

    directConnect:true,   // Only for FIreFOx and CHrome
    //seleniumAddress: 'http://localhost:4444/wd/hub/',  // For other Browsers
    SELENIUM_PROMISE_MANAGER : false,
    
    framework:'custom',
    frameworkPath:require.resolve('protractor-cucumber-framework'),
    capabilities:{
        browserName:'chrome'
    },
    specs:['../FeatureFiles/*.feature'],
    cucumberOpts:{
        tags:"",
        require:['../JSFiles/StepDefinitions/*.js'],
        format:'json:./cucumberreporter.json'
    },

    params:{                        //To Run Same Scripts in Different Environments
        //env:""              //use this - > protractor config.js --params.env="<URL>"
        letcode : "https://letcode.in/table" 

    },
    
   /* suites: {
       smoke: 'spec/smoketests/\*.js',
       full: 'spec/\*.js'   },
   */

    onPrepare:()=>{
       browser.driver.manage().window().maximize();
    },

    // onComplete:()=>{

    //     Reporter.createHTMLReport();
    // }

    onComplete:()=>{   // Cucumber HTML Reporter

        var options={
            theme:"bootstrap",
            jsonFile:'./cucumberreporter.json',
            output:'./OutputHTMLreport/Automation_Report-' +timestamp+'.html',
            reportSuiteAsScenarios:true,
            scenarioTimestamp:true,
            launchReport:true,
            metadata:{
                "App Version":"Ajin Automation 1.0",
                "Test Environment":"UAT",
                "Browser":"Chrome",
                "Platform":"",
                "Executed":""
            }
        };
        reporter.generate(options);
    }
};

**Comment by Avaddon-tfa** [comment]
Hey there! We are currently working on JS integration examples, so stay tuned, and we'll provide a solution soon.

**Reply by jobby-tech** [reply]
Hey Avaddon, Thank you , much appreciated. I have been looking for a solution with no luck....  Awaiting your inputs. Thanks again :) 

- From Across the Globe

**Reply by illiazo** [reply]
Hi, is there any news about it?

**Reply by baev** [reply]
Protractor is deprecated and will reach end-of-life in the summer of 2023. We looked into Protractor multiple times and found no way to easily integrate it with Allure. Considering deprecation I don't think we'll provide official support to it
